### ISSUE TITLE
Adding a `sampling_GQ` argument to `sample()` (feature request & design discussion)

### ISSUE THREAD

**Issue by mike-lawrence** [stan-dev/cmdstanr#428]
Similar to the FR to have a `warmup_GQ` argument, it would be useful to have an argument to turn on/off the running of the GQ section during sampling. For example, when you know that GQ will take up time in compute/IO and you just want to get the model sampling first. I've also seen users that explicitly want to always do the GQ separately, but still want the simplicity of having a GQ section in the same stanfile as the model. While there are hacks to achieve this whereby that section is commented out, etc, it seems trivial to add to cmdstanr properly, especially if we go forward with the `warmup_GQ` argument anyway. 

In contrast with the `warmup_GQ=FALSE` default I recommended in that FR, I think the default for `sampling_GQ` should be `TRUE`.

**Comment by jgabry** [comment]
> I've also seen users that explicitly want to always do the GQ separately, but still want the simplicity of having a GQ section in the same stanfile as the model.

Yeah this is related to https://github.com/stan-dev/stan/issues/2934. In general I don't think standalone generated quantities should require a separate Stan file, but I think that's something that would be preferable to implement for all Stan users not just CmdStanR users. That said, if for some reason it's decided that this won't be changed in Stan/CmdStan then we could potentially do it just in CmdStanR because I definitely agree this would be useful.

**Comment by jgabry** [comment]
Doing some old issue cleanup and I'm going to close this because I don't think a solution in CmdStanR itself is the way to go. But, like for #427, I'm still in favor of Stan supporting something like this!
